### PR TITLE
[ImageGPT] Deprecate pixel_values input name to input_ids

### PR DESCRIPTION
**Pull request by patrickvonplaten**
# What does this PR do?

As discussed in https://github.com/huggingface/transformers/pull/14784 this deprecates `pixel_values` to `input_ids`.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: src/transformers/models/imagegpt/modeling_imagegpt.py [hunk 1]
```diff
@@ -695,26 +697,33 @@ def forward(
             >>> last_hidden_states = outputs.last_hidden_state
         """
 
+        if "pixel_values" in kwargs:
+            warnings.warn(
+                "The `pixel_values` argument is deprecated and will be removed in a future version, use `input_ids` instead.",
+                FutureWarning,
+            )
+            input_ids = input_ids if input_ids is not None else kwargs.pop("pixel_values")
```
Comment (sgugger) [inline src/transformers/models/imagegpt/modeling_imagegpt.py:705]: If both `input_ids` and `pixel_values` are passed, we should raise an error.

File: src/transformers/models/imagegpt/modeling_imagegpt.py [hunk 2]
```diff
@@ -986,10 +996,17 @@ def forward(
             ...    ax.imshow(img)
         """
 
+        if "pixel_values" in kwargs:
+            warnings.warn(
+                "The `pixel_values` argument is deprecated and will be removed in a future version, use `input_ids` instead.",
+                FutureWarning,
+            )
+            input_ids = input_ids if input_ids is not None else kwargs.pop("pixel_values")
```
Comment (sgugger) [inline src/transformers/models/imagegpt/modeling_imagegpt.py:1004]: Same here.

File: src/transformers/models/imagegpt/modeling_imagegpt.py [hunk 3]
```diff
@@ -1103,10 +1121,18 @@ def forward(
             >>> outputs = model(**inputs)
             >>> logits = outputs.logits
         """
+
+        if "pixel_values" in kwargs:
+            warnings.warn(
+                "The `pixel_values` argument is deprecated and will be removed in a future version, use `input_ids` instead.",
+                FutureWarning,
+            )
+            input_ids = input_ids if input_ids is not None else kwargs.pop("pixel_values")
```
Comment (sgugger) [inline src/transformers/models/imagegpt/modeling_imagegpt.py:1130]: And here.

**Review by sgugger — APPROVED** [review]
Thanks a lot for working on this!

File: src/transformers/models/imagegpt/modeling_imagegpt.py [hunk 4]
```diff
@@ -550,30 +551,30 @@ def _set_gradient_checkpointing(self, module, value=False):
 
 IMAGEGPT_INPUTS_DOCSTRING = r"""
     Args:
-        pixel_values (:obj:`torch.LongTensor` of shape :obj:`(batch_size, pixel_values_length)`):
-            :obj:`pixel_values_length` = ``sequence_length`` if :obj:`past_key_values` is ``None`` else
+        input_ids (:obj:`torch.LongTensor` of shape :obj:`(batch_size, input_ids_length)`):
```
Comment (NielsRogge) [inline src/transformers/models/imagegpt/modeling_imagegpt.py:554]:
```suggestion
        input_ids (:obj:`torch.LongTensor` of shape :obj:`(batch_size, sequence_length)`):
```

File: src/transformers/models/imagegpt/modeling_imagegpt.py [hunk 5]
```diff
@@ -550,30 +551,30 @@ def _set_gradient_checkpointing(self, module, value=False):
...
+            computed.
         attention_mask (:obj:`torch.FloatTensor` of shape :obj:`(batch_size, sequence_length)`, `optional`):
             Mask to avoid performing attention on padding token indices. Mask values selected in ``[0, 1]``:
 
             - 1 for tokens that are **not masked**,
             - 0 for tokens that are **masked**.
 
             `What are attention masks? <../glossary.html#attention-mask>`__
-        token_type_ids (:obj:`torch.LongTensor` of shape :obj:`(batch_size, pixel_values_length)`, `optional`):
+        token_type_ids (:obj:`torch.LongTensor` of shape :obj:`(batch_size, input_ids_length)`, `optional`):
```
Comment (NielsRogge) [inline src/transformers/models/imagegpt/modeling_imagegpt.py:577]:
```suggestion
        token_type_ids (:obj:`torch.LongTensor` of shape :obj:`(batch_size, sequence_length)`, `optional`):
```

**Review by NielsRogge — APPROVED** [review]
LGTM. Can you verify the slow integration test is passing in `test_modeling_imagegpt.py`?

**Comment by patrickvonplaten** [comment]
> LGTM. Can you verify the slow integration test is passing in `test_modeling_imagegpt.py`?

All tests are passing

**Pull request merged**